modules/hostap/src/supp_main.c

```diff
@@ -340,12 +337,19 @@ static void wait_for_interface_up(const char *iface_name)
 static void iface_cb(struct net_if *iface, void *user_data)
 {
 	const char *ifname = iface->if_dev->dev->name;
+	const struct device *wifi_dev = NULL;
+
+	wifi_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_wifi));
+	if (!wifi_dev) {
+		wpa_printf(MSG_ERROR, "Device instance not found");
+		return;
+	}
 
 	if (ifname == NULL) {
 		return;
 	}
 
-	if (strncmp(ifname, DEFAULT_IFACE_NAME, strlen(ifname)) != 0) {
+	if (strncmp(ifname, wifi_dev->name, strlen(ifname)) != 0) {
```

Comment:
Even though its seldom used, we need a multi-interface solution for WPA supplicant, esp. as this is all upstream code.
And as changing interface manager isn't a trivial task, IMHO, we should stick a Kconfig of allowed prefixes and then use that to select managed interfaces in CSV form, something like below. WDYT?

```
config WPA_SUPP_MGD_IFACES_PREFIX
       string "WPA supplicant managed interfaces prefix"
       default "wlan"

CONFIG_WPA_SUPP_MGD_IFACES_PREFIX="wlan,esp,nrf"
```

Reply:
>CONFIG_WPA_SUPP_MGD_IFACES_PREFIX="wlan,esp,nrf"

Board Kconfig seems to be a good place to have this configuration. Would you agree?

Reply:
Done.